### PR TITLE
Bounddist fixes

### PR DESCRIPTION
**Pull request by michaelJwilson**

### CODE REVIEW

File: desi_fields.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# NOT COMPLETE; JUST FOR TESTING
```
Comment (michaelJwilson) [inline desi_fields.py:1]: Let's agree to use

# HACK

and #TODO, that way they're easily grepable.

Comment (michaelJwilson) [reply desi_fields.py:1]: Let's also put desi_fields in ros_tools.

File: gama_fields.py [hunk 2]
```diff
@@ -0,0 +1 @@
+gama_fields = ['G9', 'G12', 'G15']
```
Comment (michaelJwilson) [inline gama_fields.py:1]: Good to have this in gama_limits instead, I'd leave fields (already defined there) for the time being to not have everything fall over.

File: findfile.py [hunk 3]
```diff
@@ -7,8 +7,19 @@
 from   astropy.table import Table, vstack
 from   delta8_limits import d8_limits
 
+import gama_fields
+
 
 fields    = ['G9', 'G12', 'G15']
+
+if survey == ‘GAMA’:
```
Comment (michaelJwilson) [inline findfile.py:15]: Let's have this logic in the findfile function to have limited scope.

File: gen_gold_lf.py [hunk 4]
```diff
@@ -52,7 +52,7 @@ def process_cat(fpath, vmax_opath, field=None, rand_paths=[]):
     else:
         rand  = None
 
-    gama_vmax = vmaxer(gama_zmax, zmin, zmax, extra_cols=['MCOLOR_0P0', 'DDPMALL_0P0_VISZ'], rand=rand)
+    gama_vmax = vmaxer(gama_zmax, zmin, zmax, extra_cols=['MCOLOR_0P0', 'DDPMALL_0P0_VISZ', 'FIELD'], rand=rand)
```
Comment (michaelJwilson) [inline gen_gold_lf.py:55]: why's this necessary?

Comment (SgmAstro) [reply gen_gold_lf.py:55]: This made it far easier to test the V/Vmax plots for each of the three fields based on the existing code.

Comment (michaelJwilson) [reply gen_gold_lf.py:55]: Ok, makes sense!

File: fillfactor.py [hunk 5]
```diff
@@ -31,20 +35,23 @@
 field  = args.field.upper()
 dryrun = args.dryrun
 prefix = args.prefix
+survey = args.survey
```
Comment (michaelJwilson) [inline fillfactor.py:38]: petty, but let's set args.survey.lower() as the norm.

File: findfile.py [hunk 6]
```diff
@@ -26,8 +27,26 @@ def gather_cat(fpaths):
...
+    elif survey == 'desi':
+        fields = desi_fields
+    else:
+        raise NotImplementedError
 
+    return fields
+        
 def findfile(ftype, dryrun=False, prefix='', field=None, utier='{utier}', survey='gama', realz=0):
+    
+    survey = survey.lower()
```
Comment (michaelJwilson) [inline findfile.py:42]: fields = fetch_fields()

File: gen_ddp_n8.py [hunk 7]
```diff
@@ -8,24 +8,30 @@
 from   astropy.table import Table, vstack
 from   scipy.spatial import KDTree
 from   delta8_limits import delta8_tier, d8_limits
-from   gama_limits import gama_field
-
+from   gama_limits   import gama_field, gama_fields
```
Comment (michaelJwilson) [inline gen_ddp_n8.py:11]: safer to drop gama_fields and desi_fields

File: gen_ddp_n8.py [hunk 8]
```diff
@@ -46,10 +52,15 @@
 # ----  Find closest matching random to inherit fill factor  ----
 # Read randoms bound_dist.
 realz  = 0
-rpaths = [os.environ['RANDOMS_DIR'] + '/{}_bd_G{}_{:d}.fits'.format(prefix, ff, realz) for ff in [9, 12, 15]]
 
-if dryrun:
-    rpaths = [rpath.replace('.fits', '_dryrun.fits') for rpath in rpaths]
+# ----
+# BE CAREFUL HERE
+rpaths = [findfile(ftype='randoms_bd', dryrun=dryrun, field=ff, survey='gama') for ff in fields]
```
Comment (michaelJwilson) [inline gen_ddp_n8.py:58]: are you leaving survey=survey for a later PR?  Also realz as an arg.

File: gen_ddp_n8.py [hunk 9]
```diff
@@ -180,7 +193,12 @@
         print('Writing {}.'.format(opath_field))
 
         # TODO:  Here we're assuming each GAMA field has 1/3. of the area.
-        to_write_field.meta['AREA'] =  to_write.meta['AREA'] / 3.    
+        # TODO:  Work out DESI area factor (or refactor)
+        if survey == 'gama':
+            to_write_field.meta['AREA'] =  to_write.meta['AREA'] / 3.
```
Comment (michaelJwilson) [inline gen_ddp_n8.py:198]: Fine as a TODO for now. 

Suggested approach is we add field_relareas to gama_limits and ros_tools., with {'R1': 1, 'R2': 1} etc.  I.e.  the relative areas to each.

File: gen_gold_lf.py [hunk 10]
```diff
@@ -100,14 +103,17 @@ def process_cat(fpath, vmax_opath, field=None, rand_paths=[]):
...
-            fpath = fpath.replace('.fits', '_dryrun.fits')
+        #fpath = os.environ['GOLD_DIR'] + '/gama_gold_ddp.fits'
+        
+        #if dryrun:
+        #    fpath = fpath.replace('.fits', '_dryrun.fits')
 
-        opath = fpath.replace('ddp', 'vmax')
+        #opath = fpath.replace('ddp', 'vmax')
+
+        fpath = findfile(ftype='ddp', dryrun=dryrun, survey='gama')
```
Comment (michaelJwilson) [inline gen_gold_lf.py:114]: same comment as earlier, survey=survey?

File: gen_gold_lf.py [hunk 11]
```diff
@@ -173,10 +186,17 @@ def process_cat(fpath, vmax_opath, field=None, rand_paths=[]):
...
+                #all_rpaths = [os.environ['RANDOMS_DIR'] + '/{}_bd_ddp_n8_G{}_0.fits'.format(prefix, ff) for ff in [9, 12, 15]]
 
-                if dryrun:
-                    all_rpaths = [_rpath.replace('.fits', '_dryrun.fits') for _rpath in all_rpaths]
+                all_rpaths = []
+                
+                #if dryrun:
+                #    all_rpaths = [_rpath.replace('.fits', '_dryrun.fits') for _rpath in all_rpaths]
 
                 all_rands = [Table.read(_x) for _x in all_rpaths]
```
Comment (michaelJwilson) [inline gen_gold_lf.py:201]: Use gather_cat here.

**Review by SgmAstro — APPROVED** [review]

**Pull request merged**